corehq/ex-submodules/casexml/apps/case/xml/parser.py

```diff
@@ -361,6 +367,19 @@ def get_index_action(self):
     def get_attachment_action(self):
         return self._filtered_action(lambda a: isinstance(a, CaseAttachmentAction))
 
+    def get_normalized_update_property_names(self):
+        changed_properties = set()
+        if self.creates_case():
+            changed_properties.update(set(self.get_create_action().raw_block.keys()))
+        if self.updates_case():
+            changed_properties.update(set(self.get_update_action().raw_block.keys()))
```

Comment:
nit: no need to create intermediate/throw-away sets here
```suggestion
            changed_properties.update(self.get_create_action().raw_block.keys())
        if self.updates_case():
            changed_properties.update(self.get_update_action().raw_block.keys())
```